### PR TITLE
Auto-grading with exact answers

### PR DESCRIPTION
**Pull request by dariusf**
Implements feature 1 in Shuqun's feature requests.

- On creating an assessment question, a UI for adding variable numbers of correct and wrong answers will be shown.
- When the question is graded, the saved answers for it are displayed. A grade will also be filled in automatically based on whether the submitted solution matches suggested answers.

### CODE REVIEW

File: app/views/assessment/general_questions/_form.html.erb [hunk 1]
```diff
@@ -5,6 +5,34 @@
...
 
+    <%= f.input :auto_graded,
+        input_html: {"data-toggle" => "collapse", "data-target" => "#auto-grading-type"} %>
+
+    <div id="auto-grading-type" class="collapse <%= @question.auto_graded? ? "in" : "out" %>">
+
+        <%= f.input :auto_grading_type,
+            as: :radio_buttons,
+            label: 'Auto-grading type',
+            collection: [ ['None', 0], ['Exact match', 1], ['Keywords', 2]],
```
Comment (dariusf) [inline app/views/assessment/general_questions/_form.html.erb]: Rails 3 doesn't support ActiveRecord.enum. What's the preferred way to express this?

**Comment by lowjoel** [comment]
This kind of makes sense, so I think it's fine. Re specifics: @allenwq is better for v1.

File: app/views/assessment/gradings/_single_question_form.html.erb [hunk 2]
```diff
@@ -45,6 +46,7 @@
         <hr>
         <div class="row-fluid">
           <div class="span10">
+            <%= render partial: 'auto_grading_exact_options', locals: {qn: qn} %>
```
Comment (allenwq) [inline app/views/assessment/gradings/_single_question_form.html.erb]: I think you only want to render this partial for `general_questions` ?

File: app/views/assessment/general_questions/_form.html.erb (hunk 1 above)
Comment (allenwq) [inline app/views/assessment/general_questions/_form.html.erb]: How about using this gem: https://github.com/lwe/simple_enum

**Comment by allenwq** [comment]
@dariusf Can you add feature tests as well (to make sure that this feature works for submission/gradings)?  And some of the existing tests were break in this PR, please take a look.

**Comment by dariusf** [comment]
Made use of `simple_enum` (had to change schema again to follow its conventions), fixed broken PDF export, added feature spec.

The spec only covers viewing saved auto-grading options. The interface for adding them involves JS without fallback.

**Comment by allenwq** [comment]
@dariusf you forgot to push `Gemfile.lock`

**Comment by dariusf** [comment]
Fixed simple_enum version, pushed Gemfile.lock. Also removed the placeholder text for keyword options (which aren't implemented as of this PR).

**Pull request merged**